src/Event/DelayedListener.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace Biig\Component\Domain\Event;
+
+use Biig\Component\Domain\Exception\InvalidDomainEvent;
+use Biig\Component\Domain\Model\DomainModel;
+
+class DelayedListener
+{
+    /**
+     * @var string
+     */
+    private $eventName;
+
+    /**
+     * @var callable
+     */
+    private $listener;
+
+    /**
+     * @var DomainEvent[]
+     */
+    private $eventStack;
+
+    /**
+     * DelayedEvent constructor.
+     *
+     * @param string   $eventName
+     * @param callable $listener
+     */
+    public function __construct(string $eventName, callable $listener)
+    {
+        $this->eventName = $eventName;
+        $this->listener = $listener;
+        $this->eventStack = [];
+    }
+
+    /**
+     * @param DomainEvent $event
+     *
+     * @throws InvalidDomainEvent
+     */
+    public function occur(DomainEvent $event)
+    {
+        $subject = $event->getSubject();
+
+        if (!is_object($subject) || !$event->getSubject() instanceof DomainModel) {
```

Comment:
This looks right. What's your point?

Reply:
What don't you use your variable ` $subject ` ?

Reply:
indeed, i'll make the little fix after the merge of this PR because it's an export (subtree) from another project and I prefer to keep commits the same between both.